codegen/src/main/java/software/amazon/awssdk/codegen/poet/builder/BaseClientBuilderClass.java

```diff
@@ -259,33 +260,12 @@ private MethodSpec finalizeServiceConfigurationMethod() {
 
         String clientConfigClassName = model.getCustomizationConfig().getServiceConfig().getClassName();
         if (StringUtils.isNotBlank(clientConfigClassName)) {
-            ClassName clientConfigClass = ClassName.bestGuess(clientConfigClassName);
-            builder.addCode("$1T.Builder c = (($1T) config.option($2T.SERVICE_CONFIGURATION)).toBuilder();" +
-                            "c.profileFile(c.profileFile() != null ? c.profileFile() : config.option($2T.PROFILE_FILE));" +
-                            "c.profileName(c.profileName() != null ? c.profileName() : config.option($2T.PROFILE_NAME));",
-                            clientConfigClass, SdkClientOption.class);
-
-            if (model.getCustomizationConfig().getServiceConfig().hasDualstackProperty()) {
-                builder.addCode("if (c.dualstackEnabled() != null) {")
-                       .addCode("    $T.validState(config.option($T.DUALSTACK_ENDPOINT_ENABLED) == null, \"Dualstack has been "
-                                + "configured on both $L and the client/global level. Please limit dualstack configuration to "
-                                + "one location.\");",
-                                Validate.class, AwsClientOption.class, clientConfigClassName)
-                       .addCode("} else {")
-                       .addCode("    c.dualstackEnabled(config.option($T.DUALSTACK_ENDPOINT_ENABLED));", AwsClientOption.class)
-                       .addCode("}");
-            }
-
-            if (model.getCustomizationConfig().getServiceConfig().hasFipsProperty()) {
-                builder.addCode("if (c.fipsModeEnabled() != null) {")
-                       .addCode("    $T.validState(config.option($T.FIPS_ENDPOINT_ENABLED) == null, \"Fips has been "
-                                + "configured on both $L and the client/global level. Please limit fips configuration to "
-                                + "one location.\");",
-                                Validate.class, AwsClientOption.class, clientConfigClassName)
-                       .addCode("} else {")
-                       .addCode("    c.fipsModeEnabled(config.option($T.FIPS_ENDPOINT_ENABLED));", AwsClientOption.class)
-                       .addCode("}");
-            }
+            mergeServiceConfiguration(builder, clientConfigClassName);
+        }
+
+        if (model.getCustomizationConfig().useGlobalEndpoint()) {
+            builder.addStatement("$T resolver = new UseGlobalEndpointResolver(config)", ClassName.get("software.amazon.awssdk.services.s3.internal"
+                                                                         + ".endpoints", "UseGlobalEndpointResolver"));
```

Comment:
TODO: currently UseGlobalEndpointResolver class is hard-coded here. We should probably define a property in the customization config instead